client/rest/src/plugins/metadata/MetadataDb.js

```diff
@@ -78,6 +80,40 @@ class MetadataDb {
 			.toArray()
 			.then(entities => Promise.resolve(this.catapultDb.sanitizer.renameIds(entities)));
 	}
+
+	async binDataByMetalId(metalId) {
+		const compositeHashes = [routeUtils.parseArgument(metal.restoreMetadataHash(metalId), 'compositeHash', 'hash256')];
+		const metadatasByCompositeHash = await this.metadatasByCompositeHash(compositeHashes);
+		const metadataEntry = metal.getMetadataEntryByCompositehash(metadatasByCompositeHash);
```

Comment:
Do you mean that in this way we make sure to throw an exception if the result is zero?
```js
async binDataByMetalId(metalId) {
	const metadatasByCompositeHash = await this.metadatasByCompositeHash([metal.extractCompositeHashFromMetalId(metalId)]);
	if (0 === metadatasByCompositeHash.length)
		throw Error(`could not get first chunk, it may mistake the metal ID: ${metalId}`);
```
i cannot throw db test as this comment.
https://github.com/symbol/symbol/pull/944#discussion_r1499027401